packages/askui-nodejs/src/execution/ui-control-client.ts

```diff
@@ -370,31 +380,21 @@ export class UiControlClient extends ApiCommands {
     command: FluentFiltersOrRelations | FluentFiltersOrRelationsGetter,
     relation: RelationsForConvenienceMethods,
     text: string,
-  ) {
-    let commando = command;
+  ): FluentFiltersOrRelations | FluentFiltersOrRelationsGetter {
     switch (relation) {
-      case 'nearestTo':
-        commando = command.nearestTo().text(text);
-        break;
       case 'leftOf':
-        commando = command.leftOf().text(text);
-        break;
+        return command.leftOf().text(text);
       case 'above':
-        commando = command.above().text(text);
-        break;
+        return command.above().text(text);
       case 'rightOf':
-        commando = command.rightOf().text(text);
-        break;
+        return command.rightOf().text(text);
       case 'below':
-        commando = command.below().text(text);
-        break;
+        return command.below().text(text);
       case 'contains':
-        commando = command.contains().text(text);
-        break;
+        return command.contains().text(text);
       default:
-        throw new ValidationError('No valid Relation.Type was passed.');
+        return command.nearestTo().text(text);
```

Comment:
Why not
```typescript
      case 'nearestTo':
        return command.nearestTo().text(text);
```
to be explicit. 

Reasoning: If the user passes a value that is not allowed (not using TS, at least not properly), I would rather tell the user that there is something wrong with his input by failing instead of creating the assumption that this actually works. Also validation of user input should be done as early as possible. If possible, this is immediately in the first lines of the method called by the user so that in the call stack starting from there you can rely fully on typescript and the type system. There are cases where this is not possible or at least not convenient, especially for more semantical errors, e.g., need to call the database to check if an id is actually unique etc., in which cases you fail later but this is not one of such.

Reply:
I agree, because we require the user to set a correct relation type 😃 .